src/MoveBall.js

```diff
@@ -0,0 +1,163 @@
+import { Grid } from "@mui/material";
+import React, { useEffect, useState } from "react";
+import { Board } from "./Board";
+import { areBallObjectsEqual } from "./utils"
+
+const MoveBall = React.forwardRef((props, ref) => {
+
+    const RIGHT_ARROW = 39;
+    const LEFT_ARROW = 37;
+    const DOWN_ARROW = 40;
+    const A_KEYCODE = 65;
+    const S_KEYCODE = 83;
+    const D_KEYCODE = 68;
+
+    const INITIAL_VALUE = Math.ceil(props.colSize / 2) - 1;
+
+    let playerOneBall = { isPlayerOneBall: true, isPlayerTwoBall: false };
+    let playerTwoBall = { isPlayerOneBall: false, isPlayerTwoBall: true };
+
+    const [ballPlayer, setBallPlayer] = useState(playerOneBall);
+
+    const playerMovement = {
+        playerOne: {
+            rightMove: RIGHT_ARROW,
+            leftMove: LEFT_ARROW,
+            downMove: DOWN_ARROW
+        },
+        playerTwo: {
+            rightMove: D_KEYCODE,
+            leftMove: A_KEYCODE,
+            downMove: S_KEYCODE
+        }
+    };
+
+    useEffect(() => {
+        if ((ref || {}).current) {
+            (ref || {}).current.focus();
+        };
+    }, []);
+
+    useEffect(() => {
+        if (props.shouldStop) {
+            const copy = [...array];
+            const currentBallPosition = getCurrentBallPosition();
+            copy[currentBallPosition] = { isPlayerOneBall: false, isPlayerTwoBall: false };
+
+            setArray(copy);
+        };
+
+        if (props.shouldRestartGame) {
+            const copy = [...array];
+            const currentBallPosition = INITIAL_VALUE;
+
+            if (areBallObjectsEqual(props.winnerBall, playerOneBall)) {
+                copy[currentBallPosition] = { isPlayerOneBall: false, isPlayerTwoBall: true };
+            } else {
+                copy[currentBallPosition] = { isPlayerOneBall: true, isPlayerTwoBall: false };
+            };
+
+            setArray(copy);
+
+            props.afterGameRestart(false);
+        };
+    }, [props.shouldStop, props.shouldRestartGame]);
+
+    const getInitialValue = (ball) => {
```

Comment:
getInitialValue -> use a name that provides more context for what you are getting. Same for arrayOfObjects